lib/ui/views/login_view.dart

```diff
@@ -61,70 +105,112 @@ class _Container extends ProviderWidget<LoginViewModel> {
   }
 }
 
-class _EmailTextField extends ProviderWidget<LoginViewModel> {
+class _SignInButton extends ProviderWidget<LoginViewModel> {
   @override
   Widget build(BuildContext context, LoginViewModel model) {
     final local = AppLocalizations.of(context);
+    final theme = Theme.of(context);
 
-    return TextFormField(
-      controller: model.emailController,
-      validator: model.validateEmail,
-      onFieldSubmitted: (_) => model.passwordFocusNode.requestFocus(),
-      textInputAction: TextInputAction.next,
-      keyboardType: TextInputType.emailAddress,
-      decoration: InputDecoration(
-        hintText: local.emailHintText,
-        contentPadding: const EdgeInsets.all(8),
-        border: OutlineInputBorder(
-          borderRadius: const BorderRadius.all(
-            Radius.circular(12),
-          ),
-        ),
-      ),
-    );
+    return model.state == ViewState.Busy
+        ? LoadingAnimation()
+        : PlatformButton(
+            child: Text(local.loginButtonText),
+            onPressed: model.login,
+            android: (context) => MaterialRaisedButtonData(
+              textTheme: ButtonTextTheme.primary,
+              color: theme.primaryColor,
+            ),
+          );
   }
 }
 
-class _PasswordTextField extends ProviderWidget<LoginViewModel> {
+class _InputField extends ProviderWidget<LoginViewModel> {
```

Comment:
Sorry you had to work so hard on creating a platform input field. But there is a [PlatformTextField](https://github.com/aqwert/flutter_platform_widgets/blob/master/README.md#platformtextfield)